File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        List<Integer> mvp = new ArrayList<>();
        int[] answer = new int[score.length];
        for(int i = 0; i < answer.length; i++){
            int num = score[i];
            if(i < k){
                mvp.add(score[i]);
            }else{
                if(mvp.get(0) < num){
                    mvp.set(0, num);
                }
            }
            Collections.sort(mvp);
            answer[i] = mvp.get(0);
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.swift
import Foundation

func solution(_ k:Int, _ score:[Int]) -> [Int] {
    var result = Array<Int>()
    var ranking = Array<Int>()

    for i in score {
        if ranking.count >= k {
            if ranking[0] < i {
                ranking[0] = i
            }
        } else {
            ranking.append(i)
        }
        ranking.sort()
        result.append(ranking[0])
    }
    return result
}